### PR TITLE
Handle remove_domain in mod_inbox

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses "tenant removal epic".

Proposed changes include:
* Reorder indexes in the inbox table
* Add a prepared query for removal
* Add host_type type

### CODE REVIEW

File: doc/developers-guide/Testing-MongooseIM.md [hunk 1]
```diff
@@ -108,6 +108,30 @@ The command can be used instead:
 And `mam` is used to run `mam_SUITE` suite only.
 
 
+## Debugging big tests database
```
Comment (DenysGonchar) [inline doc/developers-guide/Testing-MongooseIM.md:111]: absolutely love it.
it's needed probably twice a year but always time consuming to recall how to do it. thanks a lot

File: src/inbox/mod_inbox.erl [hunk 2]
```diff
@@ -267,6 +271,12 @@ remove_user(Acc, User, Server) ->
     mod_inbox_utils:clear_inbox(User, Server),
     Acc.
 
+-spec remove_domain(mongoose_acc:t(), mongooseim:host_type(), jid:lserver()) ->
```
Comment (DenysGonchar) [inline src/inbox/mod_inbox.erl:274]: it's not `mongoose_acc:t()`, MB we need some special type to highlight that Acc is not used!

File: src/mongooseim.erl [hunk 3]
```diff
@@ -15,6 +15,8 @@
 %%==============================================================================
 -module(mongooseim).
 
+-type host_type() :: binary().
```
Comment (DenysGonchar) [inline src/mongooseim.erl:18]: 👍

**Review by DenysGonchar — APPROVED** [review]
Looks good, just one minor comment regarding Acc type. But it can be changed later in the following PRs

File: src/inbox/mod_inbox.erl (hunk 2 above)
Comment (arcusfelis) [reply src/inbox/mod_inbox.erl]: ok, changed.

**Pull request merged**